rectools/metrics/debias.py

```diff
@@ -0,0 +1,166 @@
+                f"but `{name_data}` is not assumed to be de-biased. "
+                f"Please make de-biasing for `{name_data}` "
+                "and specify `is_debiased` as `True` "
+                "or otherwise use `calc` and `calc_per_user` methods for auto de-biasing."
+            )
+
+    def make_debias(self, interactions: pd.DataFrame) -> pd.DataFrame:
+        """
+        Downsample the size of interactions, excluding some interactions with popular items.
+
+        Algorithm:
+
+            1. Calculate item "popularity"
+            (here: number of unique users that had interaction with the item) distribution from interactions;
+            2. Find first (Q1) and third (Q3) quartiles in items "popularity" distribution;
+            3. Calculate IQR = Q3 - Q1;
+            4. Calculate maximum value inside by formula: Q3 + iqr_coef * IQR;
+            5. Down-sample for all exceeding items in interactions,
+            randomly keeping the maximum group of users to a size not exceeding
+            maximum value inside
+
+        Parameters
+        ----------
+        interactions : pd.DataFrame
+            Table with previous user-item interactions,
+            with columns `Columns.User`, `Columns.Item`.
+        iqr_coef : float, default 1.5
+            Coefficient for defining as the maximum value inside the border.
+        random_state: float, default 32
+            Pseudorandom number generator state to control the down-sampling.
+
+        Returns
+        -------
+        pd.DataFrame
+            Downsampling interactions.
+        """
+        if len(interactions) == 0:
+            return interactions
+
+        interactions_for_debiasing = interactions.copy()
+
+        num_users_interacted_with_item = interactions_for_debiasing.drop_duplicates(subset=Columns.UserItem)[
+            Columns.Item
+        ].value_counts()
+
+        quantiles = num_users_interacted_with_item.quantile(q=[0.25, 0.75])
+        q1, q3 = quantiles.loc[0.25], quantiles.loc[0.75]
+        iqr = q3 - q1
+        max_border = int(q3 + self.debias_config.iqr_coef * iqr)
+
+        item_outside_max_border = num_users_interacted_with_item[num_users_interacted_with_item > max_border].index
+
+        mask_outside_max_border = interactions_for_debiasing[Columns.Item].isin(item_outside_max_border)
+        interactions_result = interactions_for_debiasing[~mask_outside_max_border]
+        interactions_downsampling = interactions_for_debiasing[mask_outside_max_border]
+
+        interactions_downsampling = (
+            interactions_downsampling.sample(frac=1.0, random_state=self.debias_config.random_state)
+            .groupby(Columns.Item)
+            .head(max_border)
+        )
+
+        result_dfs = [interactions_result, interactions_downsampling]
+        interactions_result = pd.concat(result_dfs, ignore_index=True)
+
+        return interactions_result
+
+
+def calc_debias_for_fit_metrics(
```

Comment:
1. Let's replace `calc_debias` to smth more informative. You're not calculating debiase here actually
2. Let's remove `for_fit_metrics`, it can be used anywhere. Call functions according with what they do, do not restrict their area of usage if not nessesary